### PR TITLE
Work In Progress Design Doc

### PR DESCRIPTION
**Pull request by Justin-Locke**

### CODE REVIEW

File: DesignDoc.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
+
+# TaLE Application Design
+
+## 1. Problem Statement
+
+TaLE is an application to provide travelers with information on activities in a given city.
```
Comment (jsoderkvist) [inline DesignDoc.md]: Love the name. Curious, is it an acronym of some kind?

I'd suggest fleshing out your problem statement. From just this one sentence, it's hard for me to picture what this app is going to do and who it's users will be. (Is it an eventbrite type app where people putting on events advertise their ware? Is it a crowd-sourced forum for travelers to share interesting activities with each other?)

File: DesignDoc.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+2.  As a [user], I want to view(GET) a list of activities in a city.
+3. As a [user], I want to view(GET) a single activity.
+4.  As a [user], I want to add(PUT) comments in an Activity.
+5. As a [user], I want to delete(DEL) a personal comments on an Activity.
+6.  As a [user], I want to view(GET) my posted Activities.
+7. As a [user], I want to update(POST) my Activity (Website, Hours etc...).
+8. As a [user], I want to view a list of my Comments.
+9. As a [user], I want to view my profile.
+10. As a [user], I want to view another user profile.
+
```
Comment (jsoderkvist) [inline DesignDoc.md:26]: General comment, for user stories, we want to keep them user-focused. I'd leave implementation details (like PUT, DEL, GET) for the API endpoint documentation section of the design doc.

File: DesignDoc.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+#### 6.2.3 Delete Activity Endpoint
+- Accepts a ```DELETE``` request to ```/cityId/activities/:activityId```
+- Take the userId from cognito
+- Accepts an ActivityID and returns the corresponding Deleted ActivityModel.
+
+#### 6.2.4 Update Activity Endpoint
+- Accepts a ```PUT``` requests to ```/cityId/activities/:activityId```
+- Take the userId from cognito
+- Accepts data to update an Activity including an updated Overview, WebsiteAddress, etc... returns the updated Activity.
+- Throws ```UnaughorizedUserException``` if attempted to be updated by an unauthorized User.
```
Comment (jsoderkvist) [inline DesignDoc.md]: Typo here in `UnaughorizedUserException`?

File: DesignDoc.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+```
+
+### 7.5 `CommentsByUser` GSI Table
+```
+userId // partition key, String
+commentId // sort key, String
+```
+
+## 8. Page StoryBoard
+[WIP]
```
Comment (jsoderkvist) [inline DesignDoc.md]: I strongly encourage creating some mock-ups as your next step before going further with the technical design. I think it'll really help you clarify what endpoints and data models you'll need.

File: DesignDoc.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
+
+# TaLE Application Design
+
+## 1. Problem Statement
+
+TaLE is an application to provide travelers with information on activities in a given city.
+
+## 2. Top Questions to Resolve in Review
+
+1. How should the table between Activities and Cities be set up? Should States contain cities while cities contain activities?
```
Comment (jsoderkvist) [inline DesignDoc.md:12]: Your idea of focusing on Nashville for the MVP sounds wise to me.

File: DesignDoc.md (hunk 1 above)
Comment (Justin-Locke) [reply DesignDoc.md]: I have Updated This portion of the design doc.

File: DesignDoc.md (hunk 3 above)
Comment (Justin-Locke) [reply DesignDoc.md]: Fixed Typo

File: DesignDoc.md (hunk 4 above)
Comment (Justin-Locke) [reply DesignDoc.md]: Created Mock Up.

File: DesignDoc.md (hunk 5 above)
Comment (Justin-Locke) [reply DesignDoc.md:12]: Have made the mock up to focus on using Nashville as the first and ONLY city...to begin with.

File: DesignDoc.md [hunk 6]
```diff
@@ -175,11 +176,11 @@ List<CityIds>
 ```
 activityId // partition key, string
 userId // String
-commentIds // List of Strings of commentIds
 Description // String
 Website // String
 Location // String
 Date Posted // LocalDate
+edited // Bool
```
Comment (jsoderkvist) [inline DesignDoc.md:183]: Will you remind me, what does `edited` represent here? (How will it be used?)

I see `edited` on the CommentsByUser GSI, too. Also not recalling what that one is for, but if it's something you need in the GSI you'll need to have it on the original table, too.

File: DesignDoc.md [hunk 7]
```diff
@@ -156,16 +157,16 @@ List<CityIds>
...
 #### 6.2.9 Delete a Single Comment Endpoint
-- Accepts ```DELETE``` request to ```/comments/{commentId}```
+- Accepts ```DELETE``` request to ```/activities/{activityId}/comments/{commentId}```
 - Deletes associated comment with commentID
 
 ### 6.2.10 Get All Activities For City Endpoint
--Accepts ```GET``` request to ```/cities/{cityId]```
+-Accepts ```GET``` request to ```/cities/{cityId}```
 - Accepts the cityId and returns all Associated Activity Names with City.
 - Throw ```InvalidActivityIdException``` if the called activity Id does not exist.
```
Comment (jsoderkvist) [inline DesignDoc.md:171]: Endpoints look good to me!

File: DesignDoc.md (hunk 6 above)
Comment (jsoderkvist) [reply DesignDoc.md:183]: Oh, one other general suggestion... I'd suggest picking a convention to use for your table attributes (for example, always starting with lowercase) and using that consistently. Being consistent with things like that makes it easier to read and maintain the code later 'cause it's more predictable.

Comment (Justin-Locke) [reply DesignDoc.md:183]: I had envisioned edit could be a boolean value that shows if the Activity/Comment has been edited.

Comment (jsoderkvist) [reply DesignDoc.md:183]: Ah, okay. In that case, sounds like you'd want it on the `comments` table, too (so that you can have it on the GSI).

**Review by jsoderkvist — APPROVED** [review]
I encourage addressing the remaining open comments in the table design section (specifically, standardizing the capitalization and including the edited attribute on the comments table if you want it on the GSI). I don't consider those blockers to moving forward officially to the implementation phase, though. Let's call this design approved!

**Pull request merged**